library/coretests/tests/not_doctests.rs

```diff
@@ -646,3 +646,97 @@ int_step! {
     u128 => u128_step,
     usize => usize_step,
 }
+
+// covers:
+// - `<core::mem::Discriminant<T> as core::cmp::PartialEq>::eq`
+// - `core::mem::discriminant`
+#[test]
+fn discriminant() {
+    enum Foo {
+        A,
+        B,
+    }
+    assert_eq!(core::mem::discriminant(&Foo::A), core::mem::discriminant(&Foo::A));
+    assert_ne!(core::mem::discriminant(&Foo::A), core::mem::discriminant(&Foo::B));
+}
+
+// covers:
+// - `<core::mem::maybe_uninit::MaybeUninit<T> as core::clone::Clone>::clone`
+// - `core::mem::maybe_uninit::MaybeUninit::<T>::as_bytes`
+// - `core::mem::maybe_uninit::MaybeUninit::<T>::as_bytes_mut`
+#[test]
+fn maybe_uninit() {
+    let mut a = core::mem::MaybeUninit::<u64>::uninit();
+    let mut b = a.clone();
+
+    a.write(u64::MAX);
+    for (x, y) in a.as_bytes().into_iter().zip(b.as_bytes_mut()) {
+        let x = unsafe { x.assume_init_read() };
+        y.write(x);
+    }
+
+    let a = unsafe { a.assume_init() };
+    let b = unsafe { b.assume_init() };
+    assert_eq!(a, u64::MAX);
+    assert_eq!(b, u64::MAX);
+}
+
+// covers `core::ptr::const_ptr::<impl *const T>::align_offset`
+#[test]
+#[should_panic]
```

Comment:
nit: could you put a panic message here? I think the problem here is that 3 isn't aligned to a pointer but I'm not entirely sure.